cli/determined_cli/tunnel.py

```diff
@@ -129,14 +139,11 @@ def http_connect_tunnel(master: str, service: str, master_cert: Optional[str]) -
 
 
 if __name__ == "__main__":
-    if len(sys.argv) not in (3, 4):
-        print(
-            "usage: {} MASTER_ADDR SERVICE_UUID [MASTER_CERT]".format(sys.argv[0]), file=sys.stderr
-        )
-        sys.exit(1)
-
-    master = sys.argv[1]
-    service = sys.argv[2]
-    master_cert = None if len(sys.argv) < 4 else sys.argv[3]
-
-    http_connect_tunnel(master, service, master_cert)
+    parser = argparse.ArgumentParser(description="Tunnel through a Determined Master")
+    parser.add_argument("master_addr", type=str)
+    parser.add_argument("service_uuid", type=str)
+    parser.add_argument("--cert-file", type=str, nargs="?")
+    parser.add_argument("--cert-name", type=str, nargs="?")
```

Comment:
non-blocking: `nargs` isn't doing anything for us, is it? The options are optional, but the argument may as well be required if the option is present.

Reply:
Guess not.  Neither is `type=str`, looks like.